File: IOS/DTOs/Objects/Customer/Favorite.swift
import Foundation

public enum Favorite {
    public enum Responses {}
}

public extension Favorite.Responses {
    typealias Salons = [Salon]
    
    struct Salon: Codable, Sendable {
        public var id: UUID
        public var name: String
        public var type: SalonType
        public var logo: URL
        public var isActive: Bool
        public var address: Address
        public var point: CoordinatePoint
        
        public init(
            id: UUID,
            name: String,
            type: SalonType,
            logo: URL,
            isActive: Bool,
            address: Address,
            point: CoordinatePoint
        ) {
            self.id = id
            self.name = name
            self.type = type
            self.logo = logo
            self.isActive = isActive
            self.address = address
            self.point = point
        }
    }
}

File: Android/shared/src/main/kotlin/shared/dto/objects/Favorite.kt
@file:UseSerializers(UUIDSerializer::class, URISerializer::class)

package shared.dto.objects

import shared.dto.enums.SalonType
import shared.dto.primitives.Address
import shared.serializers.URISerializer
import java.net.URI

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.primitives.CoordinatePoint
import shared.dto.protocols.Responsable
import shared.serializers.UUIDSerializer
import java.util.*

typealias Salons = List<Salon>

object Favorite {
    data object Responses {

        @Serializable
        data class Salon(
            val id: UUID,
            val name: String,
            val type: SalonType,
            val logo: URI,
            val isActive: Boolean,
            val address: Address,
            val point: CoordinatePoint,
            ) : Responsable
    }
}
